### PR TITLE
new package: cello

### LINKED ISSUE #3678: Add Cello
### Is your feature request related to a problem? Please describe.

Cannot port [Cello](https://github.com/orangeduck/Cello) to xmake projects.

### Describe the solution you'd like

Adding Cello to xmake-repo.

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by cattokomo**
Closes #3678.

### CODE REVIEW

File: packages/c/cello/xmake.lua [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+                    if is_plat("@mingw") and target:has_cincludes("Dbghelp.h") then
+                        target:add("links", "Dbghelp")
+                    elseif is_plat("@linux") and target:has_cincludes("execinfo.h") then
+                        target:add({ links = "execinfo", ldflags = "-rdynamic" })
+                    else
+                        target:add("defines", "CELLO_NSTRACE")
+                    end
+                end)
+        ]])
+        import("package.tools.xmake").install(package, { kind = package:config("shared") and "shared" or "static" })
```
Comment (star-hengxing) [inline packages/c/cello/xmake.lua:30]: xmake will automatically pass the shared config, please remove it

https://github.com/xmake-io/xmake/blob/d8c1f612cdbd5c6f29a7b789d46f91292c1abc1e/xmake/modules/package/tools/xmake.lua#L238-L240

File: packages/c/cello/xmake.lua [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+    add_versions("2.0.3", "1afcae06f5efc10ea161737a862073ff5679c964540bca7cd719539609d0633c")
+
+    on_install("@!windows", "android@linux", function(package)
+        io.writefile("xmake.lua", [[
+            add_rules("mode.release", "mode.debug")
+            target("Cello")
+                set_kind("$(kind)")
+                set_languages("gnu99")
+                add_links("m", "pthread")
```
Comment (star-hengxing) [inline packages/c/cello/xmake.lua:16]: Please use `add_syslinks`

File: packages/c/cello/xmake.lua [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+                add_files("src/*.c")
+                add_headerfiles("include/Cello.h")
+                add_includedirs("include")
+                on_config(function(target)
+                    if is_plat("@mingw") and target:has_cincludes("Dbghelp.h") then
+                        target:add("links", "Dbghelp")
+                    elseif is_plat("@linux") and target:has_cincludes("execinfo.h") then
+                        target:add({ links = "execinfo", ldflags = "-rdynamic" })
```
Comment (star-hengxing) [inline packages/c/cello/xmake.lua:24]: and here

Comment (cattokomo) [reply packages/c/cello/xmake.lua]: Resolved in [ef9dca4](https://github.com/xmake-io/xmake-repo/pull/3679/commits/ef9dca451d67d2f176bbcfe5edd0920e6d711786).

File: packages/c/cello/xmake.lua (hunk 2 above)
Comment (cattokomo) [reply packages/c/cello/xmake.lua]: Resolved in [ef9dca4](https://github.com/xmake-io/xmake-repo/pull/3679/commits/ef9dca451d67d2f176bbcfe5edd0920e6d711786).

File: packages/c/cello/xmake.lua (hunk 1 above)
Comment (cattokomo) [reply packages/c/cello/xmake.lua]: Resolved in [ef9dca4](https://github.com/xmake-io/xmake-repo/pull/3679/commits/ef9dca451d67d2f176bbcfe5edd0920e6d711786).

**Pull request closed without merging**